Pod/Classes/LSPAudioViewController.m

```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
     {
         if ([keyPath isEqualToString:NSStringFromSelector(@selector(playing))])
         {
-            [self.view showLayoutForPlaying:self.player.isPlaying];
+            [self.playerView showLayoutForPlaying:self.player.isPlaying];
         }
     }
 }
 
 
 #pragma mark - Layout
-- (void)assignConstraintsToView:(UIView *)parentView
+- (void)assignConstraintsToView
+{
+    [self.playerView mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.left.and.right.equalTo(self.view);
+        make.height.mas_equalTo(self.configuration.height);
+        make.bottom.equalTo(self.bottomLayout);
+    }];
+    [self.view layoutIfNeeded];
+}
+
+
+- (void)show
 {
-    NSLayoutConstraint *heightConstraint = [NSLayoutConstraint constraintWithItem:self.view attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:0 multiplier:1 constant:60];
-    [self.view addConstraint:heightConstraint];
+    CGFloat offset = [self bottomOffset];
+    [self.playerView mas_updateConstraints:^(MASConstraintMaker *make) {
+        make.bottom.equalTo(self.bottomLayout).with.offset(offset);
+    }];
+    [self.view layoutIfNeeded];
     
-    self.bottomConstraint = [NSLayoutConstraint constraintWithItem:self.view attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeBottom multiplier:1 constant:0];
-    [parentView addConstraint:self.bottomConstraint];
+    NSTimeInterval duration = .666f;
+    NSTimeInterval delay = 0;
+    UIViewAnimationOptions options = (UIViewAnimationOptionBeginFromCurrentState | UIViewAnimationOptionCurveEaseInOut);
     
-    UIView *view = self.view;
-    NSDictionary *viewBindings = NSDictionaryOfVariableBindings(view);
-    [parentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[view]|" options:0 metrics:nil views:viewBindings]];
+    __weak typeof(self) weakSelf = self;
+    [UIView animateWithDuration:duration delay:delay options:options animations:^{
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        [strongSelf.playerView mas_updateConstraints:^(MASConstraintMaker *make) {
+            make.bottom.equalTo(self.bottomLayout).with.offset(0);
+        }];
+        [strongSelf.view layoutIfNeeded];
+    } completion:nil];
+}
+
+
+- (void)hide:(nullable void (^)(void))completion
+{
+    __weak typeof(self) weakSelf = self;
+    [weakSelf.view layoutIfNeeded];
+    
+    NSTimeInterval duration = .666f;
+    NSTimeInterval delay = 0;
+    UIViewAnimationOptions options = (UIViewAnimationOptionBeginFromCurrentState | UIViewAnimationOptionCurveEaseInOut);
+    
+    [UIView animateWithDuration:duration delay:delay options:options animations:^{
```

Comment:
I think no reference cycles can appear in this block, so `weakify` is redundant here.